galsim/sed.py

```diff
@@ -818,6 +818,49 @@ def calculateSeeingMomentRatio(self, bandpass, alpha=-0.2, base_wavelength=500):
             return galsim.integ.int1d(lambda w: weight(w) * kernel(w),
                                       bandpass.blue_limit, bandpass.red_limit) / flux
 
+    def sampleWavelength(self, nphotons, bandpass, seed=0, npoints=256):
+        """ Sample 'nphoton' random wavelengths from the SED (and bandpass).
+
+        @param nphotons  Number of samples (photons) to randomly draw.
+        @param bandpass  Bandpass through which object is being imaged.
+        @param seed      Something that can seed a BaseDeviate: a long int seed or another
```

Comment:
You can see an example of how this is done in real.py, in the RealGalaxy constructor.